### PR TITLE
fix: can not modify x-forwarded-host in proxy rewrite

### PR DESCRIPTION
**Pull request by monkeyDluffy6017**
### Description

fix can not modify x-forwarded-host in proxy rewrite

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [ ] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

File: t/plugin/proxy-rewrite.t [hunk 1]
```diff
@@ -1311,3 +1311,56 @@ host: test.com:6443
 x-real-ip: 127.0.0.1
 --- no_error_log
 [error]
+
+
+
+=== TEST 45: set route(rewrite host)
+--- config
```
Comment (spacewander) [inline t/plugin/proxy-rewrite.t:1318]: Let's add new test to https://github.com/apache/apisix/blob/master/t/plugin/proxy-rewrite3.t

File: t/plugin/proxy-rewrite.t [hunk 2]
```diff
@@ -1311,3 +1311,56 @@ host: test.com:6443
 x-real-ip: 127.0.0.1
 --- no_error_log
 [error]
+
+
+
+=== TEST 45: set route(rewrite host)
```
Comment (spacewander) [inline t/plugin/proxy-rewrite.t:1317]: The title is incorrect. We don't rewrite the host

File: t/plugin/proxy-rewrite.t (hunk 1 above)
Comment (monkeyDluffy6017) [reply t/plugin/proxy-rewrite.t]: Done

File: t/plugin/proxy-rewrite.t (hunk 2 above)
Comment (monkeyDluffy6017) [reply t/plugin/proxy-rewrite.t]: Done

File: t/plugin/proxy-rewrite.t [hunk 3]
```diff
@@ -34,7 +34,61 @@ run_tests;
 
 __DATA__
 
-=== TEST 1: sanity
+=== TEST 1: set route(rewrite host)
```
Comment (tokers) [inline t/plugin/proxy-rewrite.t:37]: This case is samiliar to the one in proxy-rewrite-3.t?

Comment (monkeyDluffy6017) [reply t/plugin/proxy-rewrite.t]: Done

**Review by spacewander — APPROVED** [review]

**Review by tzssangglass — APPROVED** [review]

**Pull request merged**